evaluate_iteration_changes.py

```diff
@@ -56,12 +56,14 @@ def collect_iteration_stats(
                 current_iteration_stats["summary"][sum_id] = {
                     "summary": data["summary"]
                 }
+                failed_sums_by_id[sum_id] = iteration_idx
```

Comment:
it doesn't seem like there's anything here to specify to failed summaries..why does this work?

Reply:
What do you mean? It sets the sum_id in the dict so you can just look up whether the sum_id is in the dict to determine that a summary failed

Reply:
I'm likely just missing it, but there's nothing specific to `summaries` on L51, L44 that means that all the summaries are failed, right? I presume that `summaries = iteration_data["summaries"]` includes much more than failed summaries...